packages/editor-ui/src/Interface.ts

```diff
@@ -1269,3 +1269,22 @@ export interface CurlToJSONResponse {
 	"parameters.sendQuery": boolean;
 	"parameters.sendBody": boolean;
 }
+
+export type Basic = string | number | boolean;
+export type Primitives = Basic | bigint | symbol;
+
+export type Optional<T> = T | undefined | null;
+
+export type N8nJsonSchemaType =
+	| 'string'
+	| 'number'
+	| 'boolean'
+	| 'bigint'
+	| 'symbol'
+	| 'date'
+	| 'list'
+	| 'object'
+	| 'function'
+	| 'null'
+	| 'undefined';
+export type N8nJsonSchema = { type: N8nJsonSchemaType, key?: string, value: string | N8nJsonSchema[], path?: string };
```

Comment:
why is path optional? and key? is it because of the first layer? If so, can we rewrite these are for only first layer are not needed? 
```
type Schema = { type: N8nJsonSchemaType, key: string, value: string | Schema[], path: string };
export type JsonSchema = { type: N8nJsonSchemaType, value: string | Schema[] };
```

Reply:
I have made `path` required. It should be like that anyways because how could you select something from JSON if you don't pass a path (even the empty string which means select the root/everything)

About `key`: the only exception here is `list` (more precisely list of lists)
In the case of a list of lists the schema has its `value` of `Schema[]` but with no `key`